### PR TITLE
Improved types, README documentation, simplify and extend unit tests

### PR DESCRIPTION
**Pull request by nalundgaard**
* Remove obsolete `deps` target from Makefile.
* Use a single unified eunit test generator for each module's tests, to reduce repetition.
* Leverage the `setup/0` function for storing common test parameters.
* Add tests of invalid tokens in `pot:valid_hotp` and `pot:valid_totp` to achieve full line coverage in unit tests.
* Extend hotp_generation_tests to reason about 0 and negative/very large interval (counter) values.
* Improve type specifications for module parameters, particularly options.
* Rename `time()` to `interval()` for clarity.
* Export types for clients to reference as-needed.
* Remove invalid `casefold` option in `hotp` function (this parameter is not set correctly and is unsupported by underlying base32 decoder, which is already case-insensitive).
* Re-order library functions to group token generation and token validation.
* Update README with TOC, cross-references, links table at the bottom, and function references.

### CODE REVIEW

File: test/hotp_validity_tests.erl [hunk 1]
```diff
@@ -11,6 +11,7 @@ checking_hotp_validity_without_range_test_() ->
      [fun checking_hotp_validity_without_range/1,
       fun checking_hotp_validity_max_default_range/1,
       fun checking_hotp_validity_past_max_default_range/1,
+      fun validatinging_invalid_token_hotp/1,
```
Comment (va17) [inline test/hotp_validity_tests.erl]: s/validatinging/validating

File: test/hotp_validity_tests.erl [hunk 2]
```diff
@@ -40,6 +41,10 @@ checking_hotp_validity_past_max_default_range(Secret) ->
       ?_assertNot(pot:valid_hotp(pot:hotp(Secret, 1002), Secret))}].
 
 
+validatinging_invalid_token_hotp(Secret) ->
```
Comment (va17) [inline test/hotp_validity_tests.erl]: s/validatinging/validating

File: test/totp_validity_tests.erl [hunk 3]
```diff
@@ -9,36 +9,37 @@ totp_validity_test_() ->
...
 
-validating_invalid_totp_for_same_secret(_) ->
-    Secret = <<"MFRGGZDFMZTWQ2LK">>,
+validating_invalid_totp_for_same_secret(Secret) ->
     [?_assertNot(pot:valid_totp(<<"123456">>, Secret))].
 
 
-validating_correct_hotp_as_totp(_) ->
-    Secret = <<"MFRGGZDFMZTWQ2LK">>,
+validatinging_invalid_token_hotp(Secret) ->
```
Comment (va17) [inline test/totp_validity_tests.erl]: s/validatinging/validating

File: test/totp_validity_tests.erl [hunk 4]
```diff
@@ -9,36 +9,37 @@ totp_validity_test_() ->
      fun stop/1,
      [fun validating_totp_for_same_secret/1,
       fun validating_invalid_totp_for_same_secret/1,
+      fun validatinging_invalid_token_hotp/1,
```
Comment (va17) [inline test/totp_validity_tests.erl]: s/validatinging/validating

**Review by va17 — COMMENTED** [review]
LGTM. Thanks a lot for README update! It will make `pot` learning curve faster for new users :)

**Comment by nalundgaard** [comment]
Thanks @va17, the spelling error should be fixed now.

**Review by va17 — APPROVED** [review]

**Comment by yuce** [comment]
@nalundgaard  Wow, this PR is great, thanks! Do you mind if I add you as a collaborator to the project so you can merge PRs, etc.?

**Comment by nalundgaard** [comment]
@yuce Thanks! I'm happy to contribute back—we've really appreciated this project at my company. I hope more accessible docs will encourage further adoption. :)

I'd be happy to work with you on the project as a collaborator, too.

**Comment by yuce** [comment]
Fantastic, thanks! I've added you as a collaborator. Looking forward to more of your contributions! :)

**Pull request merged**